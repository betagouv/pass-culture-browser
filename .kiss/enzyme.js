import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
