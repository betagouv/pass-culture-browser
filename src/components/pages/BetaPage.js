/* eslint
  react/jsx-one-expression-per-line: 0 */
import React from 'react'
import { Link } from 'react-router-dom'

const BetaPage = () => (
  <div id="beta-page" className="page pc-gradient flex-rows">
    <main role="main" className="pc-main padded flex-rows flex-center flex-0">
      <h1 className="text-left fs32">
        <span className="is-bold is-italic is-block">Bienvenue</span>
        <span className="is-italic is-block">dans la version beta</span>
        <span className="is-italic is-block">du Pass Culture</span>
      </h1>
      <p className="text-left mt36 fs22">
        <span className="is-block">Et merci de votre participation</span>
        <span className="is-block">pour nous aider à l&apos;améliorer !</span>
      </p>
    </main>
    <footer role="navigation" className="pc-footer flex-columns flex-end">
      <Link to="/inscription" className="flex-center items-center">
        <span className="fs32 is-italic is-bold">C&apos;est par là</span>
        <span
          aria-hidden="true"
          className="pc-icon icon-next-long"
          title="C&apos;est par là"
        />
      </Link>
    </footer>
  </div>
)

export default BetaPage
