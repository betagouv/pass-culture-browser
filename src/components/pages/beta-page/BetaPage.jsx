import React from 'react'
import FormFooter from '../../forms/FormFooter'
import { ICONS_URL } from '../../../utils/config'

const BetaPage = () => (
  <div className="beta-page">
    <img
      alt=""
      className="bp-logo"
      src={`${ICONS_URL}/circle.svg`}
    />
    <main className="bp-main">
      <div className="bp-title">
        {'Bienvenue dans\n'}
        {'votre pass Culture'}
      </div>
      <div className="bp-content">
        {'Vous avez 18 ans et vivez dans un\n'}
        <a
          href='https://pass.culture.fr/#jeune'
          rel="noopener noreferrer"
          target="_blank"
        >
          {'département éligible ?'}
        </a>
      </div>
      <div className="bp-content">
        {'Bénéficiez de 500 € afin de\n'}
        {'renforcer vos pratiques\n'}
        {"culturelles et d'en découvrir\n"}
        {'de nouvelles !'}
      </div>
    </main>
    <div className="bp-buttons-container">
      <FormFooter
        externalLink={{
          id: 'sign-up-link',
          label: 'Créer un compte',
          target: '_blank',
          url: 'https://www.demarches-simplifiees.fr/commencer/inscription-pass-culture',
        }}
        submit={{
          id: 'sign-in-link',
          label: "J'ai un compte",
          url: '/connexion'
        }}
      />
    </div>
  </div>
)

export default BetaPage
