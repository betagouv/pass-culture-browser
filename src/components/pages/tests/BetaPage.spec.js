import React from 'react'
import { shallow } from 'enzyme'

import BetaPage from '../BetaPage'

describe('src | components | pages | BetaPage', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        errors: {},
      }

      // when
      const wrapper = shallow(<BetaPage {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
