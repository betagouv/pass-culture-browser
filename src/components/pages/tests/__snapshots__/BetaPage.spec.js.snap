// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | BetaPage snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BetaPage
    errors={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <main
          className="pc-main flex-rows flex-center flex-0"
          role="main"
        >
          <h1
            className="text-left fs32"
          >
            <b>
              <i
                className="is-block"
              >
                Bienvenue
              </i>
            </b>
            <i
              className="is-block"
            >
              dans la version beta
            </i>
            <i
              className="is-block"
            >
              du Pass Culture
            </i>
          </h1>
          <p
            className="text-left mt36 fs22"
          >
            <i
              className="is-block"
            >
              Et merci de votre participation
            </i>
            <i
              className="is-block"
            >
              pour nous aider à l'améliorer !
            </i>
          </p>
        </main>,
        <footer
          className="pc-footer flex-columns flex-end"
          role="navigation"
        >
          <Link
            className="flex-center items-center"
            replace={false}
            to="/inscription"
          >
            <b
              className="fs32"
            >
              <i>
                C'est par là
              </i>
            </b>
            <span
              aria-hidden="true"
              className="pc-icon icon-next-long"
              title="C'est par là"
            />
          </Link>
        </footer>,
      ],
      "className": "page pc-gradient flex-rows",
      "id": "beta-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="text-left fs32"
            >
              <b>
                <i
                  className="is-block"
                >
                  Bienvenue
                </i>
              </b>
              <i
                className="is-block"
              >
                dans la version beta
              </i>
              <i
                className="is-block"
              >
                du Pass Culture
              </i>
            </h1>,
            <p
              className="text-left mt36 fs22"
            >
              <i
                className="is-block"
              >
                Et merci de votre participation
              </i>
              <i
                className="is-block"
              >
                pour nous aider à l'améliorer !
              </i>
            </p>,
          ],
          "className": "pc-main flex-rows flex-center flex-0",
          "role": "main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  <i
                    className="is-block"
                  >
                    Bienvenue
                  </i>
                </b>,
                <i
                  className="is-block"
                >
                  dans la version beta
                </i>,
                <i
                  className="is-block"
                >
                  du Pass Culture
                </i>,
              ],
              "className": "text-left fs32",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="is-block"
                  >
                    Bienvenue
                  </i>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Bienvenue",
                    "className": "is-block",
                  },
                  "ref": null,
                  "rendered": "Bienvenue",
                  "type": "i",
                },
                "type": "b",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "dans la version beta",
                  "className": "is-block",
                },
                "ref": null,
                "rendered": "dans la version beta",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "du Pass Culture",
                  "className": "is-block",
                },
                "ref": null,
                "rendered": "du Pass Culture",
                "type": "i",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="is-block"
                >
                  Et merci de votre participation
                </i>,
                <i
                  className="is-block"
                >
                  pour nous aider à l'améliorer !
                </i>,
              ],
              "className": "text-left mt36 fs22",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Et merci de votre participation",
                  "className": "is-block",
                },
                "ref": null,
                "rendered": "Et merci de votre participation",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "pour nous aider à l'améliorer !",
                  "className": "is-block",
                },
                "ref": null,
                "rendered": "pour nous aider à l'améliorer !",
                "type": "i",
              },
            ],
            "type": "p",
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="flex-center items-center"
            replace={false}
            to="/inscription"
          >
            <b
              className="fs32"
            >
              <i>
                C'est par là
              </i>
            </b>
            <span
              aria-hidden="true"
              className="pc-icon icon-next-long"
              title="C'est par là"
            />
          </Link>,
          "className": "pc-footer flex-columns flex-end",
          "role": "navigation",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <b
                className="fs32"
              >
                <i>
                  C'est par là
                </i>
              </b>,
              <span
                aria-hidden="true"
                className="pc-icon icon-next-long"
                title="C'est par là"
              />,
            ],
            "className": "flex-center items-center",
            "replace": false,
            "to": "/inscription",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i>
                  C'est par là
                </i>,
                "className": "fs32",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "C'est par là",
                },
                "ref": null,
                "rendered": "C'est par là",
                "type": "i",
              },
              "type": "b",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "pc-icon icon-next-long",
                "title": "C'est par là",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <main
            className="pc-main flex-rows flex-center flex-0"
            role="main"
          >
            <h1
              className="text-left fs32"
            >
              <b>
                <i
                  className="is-block"
                >
                  Bienvenue
                </i>
              </b>
              <i
                className="is-block"
              >
                dans la version beta
              </i>
              <i
                className="is-block"
              >
                du Pass Culture
              </i>
            </h1>
            <p
              className="text-left mt36 fs22"
            >
              <i
                className="is-block"
              >
                Et merci de votre participation
              </i>
              <i
                className="is-block"
              >
                pour nous aider à l'améliorer !
              </i>
            </p>
          </main>,
          <footer
            className="pc-footer flex-columns flex-end"
            role="navigation"
          >
            <Link
              className="flex-center items-center"
              replace={false}
              to="/inscription"
            >
              <b
                className="fs32"
              >
                <i>
                  C'est par là
                </i>
              </b>
              <span
                aria-hidden="true"
                className="pc-icon icon-next-long"
                title="C'est par là"
              />
            </Link>
          </footer>,
        ],
        "className": "page pc-gradient flex-rows",
        "id": "beta-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="text-left fs32"
              >
                <b>
                  <i
                    className="is-block"
                  >
                    Bienvenue
                  </i>
                </b>
                <i
                  className="is-block"
                >
                  dans la version beta
                </i>
                <i
                  className="is-block"
                >
                  du Pass Culture
                </i>
              </h1>,
              <p
                className="text-left mt36 fs22"
              >
                <i
                  className="is-block"
                >
                  Et merci de votre participation
                </i>
                <i
                  className="is-block"
                >
                  pour nous aider à l'améliorer !
                </i>
              </p>,
            ],
            "className": "pc-main flex-rows flex-center flex-0",
            "role": "main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    <i
                      className="is-block"
                    >
                      Bienvenue
                    </i>
                  </b>,
                  <i
                    className="is-block"
                  >
                    dans la version beta
                  </i>,
                  <i
                    className="is-block"
                  >
                    du Pass Culture
                  </i>,
                ],
                "className": "text-left fs32",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="is-block"
                    >
                      Bienvenue
                    </i>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Bienvenue",
                      "className": "is-block",
                    },
                    "ref": null,
                    "rendered": "Bienvenue",
                    "type": "i",
                  },
                  "type": "b",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "dans la version beta",
                    "className": "is-block",
                  },
                  "ref": null,
                  "rendered": "dans la version beta",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "du Pass Culture",
                    "className": "is-block",
                  },
                  "ref": null,
                  "rendered": "du Pass Culture",
                  "type": "i",
                },
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="is-block"
                  >
                    Et merci de votre participation
                  </i>,
                  <i
                    className="is-block"
                  >
                    pour nous aider à l'améliorer !
                  </i>,
                ],
                "className": "text-left mt36 fs22",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Et merci de votre participation",
                    "className": "is-block",
                  },
                  "ref": null,
                  "rendered": "Et merci de votre participation",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "pour nous aider à l'améliorer !",
                    "className": "is-block",
                  },
                  "ref": null,
                  "rendered": "pour nous aider à l'améliorer !",
                  "type": "i",
                },
              ],
              "type": "p",
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="flex-center items-center"
              replace={false}
              to="/inscription"
            >
              <b
                className="fs32"
              >
                <i>
                  C'est par là
                </i>
              </b>
              <span
                aria-hidden="true"
                className="pc-icon icon-next-long"
                title="C'est par là"
              />
            </Link>,
            "className": "pc-footer flex-columns flex-end",
            "role": "navigation",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <b
                  className="fs32"
                >
                  <i>
                    C'est par là
                  </i>
                </b>,
                <span
                  aria-hidden="true"
                  className="pc-icon icon-next-long"
                  title="C'est par là"
                />,
              ],
              "className": "flex-center items-center",
              "replace": false,
              "to": "/inscription",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i>
                    C'est par là
                  </i>,
                  "className": "fs32",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "C'est par là",
                  },
                  "ref": null,
                  "rendered": "C'est par là",
                  "type": "i",
                },
                "type": "b",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "pc-icon icon-next-long",
                  "title": "C'est par là",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
